src/main/java/klaxon/klaxon/novisoculis/CubeModel.java

```diff
@@ -0,0 +1,207 @@
+    private static final int[][] vertices = {
+        { 0, 0, 0 },
+        { 0, 0, 1 },
+        { 1, 0, 1 },
+        { 1, 0, 0 },
+        { 0, 1, 0 },
+        { 0, 1, 1 },
+        { 1, 1, 1 },
+        { 1, 1, 0 }
+    };
+
+    private static final int[][] uv = {
+        { 0, 0 },
+        { 0, 1 },
+        { 1, 1 },
+        { 1, 0 }
+    };
+
+    // Add the UV.
+    private static final int[][][] shiftsByDirection = {
+
+        // DOWN 1 0 3 2
+        {
+            vertices[1],
+            vertices[0],
+            vertices[3],
+            vertices[2]
+        },
+
+        //UP 4 5 6 7
+        {
+            vertices[4],
+            vertices[5],
+            vertices[6],
+            vertices[7]
+        },
+
+        //NORTH 7 3 0 4
+        {
+            vertices[7],
+            vertices[3],
+            vertices[0],
+            vertices[4]
+        },
+
+        //SOUTH 5 1 2 6
+        {
+            vertices[5],
+            vertices[1],
+            vertices[2],
+            vertices[6]
+        },
+
+        //WEST 4 0 1 5
+        {
+            vertices[4],
+            vertices[0],
+            vertices[1],
+            vertices[5]
+        },
+
+        //EAST 6 2 3 7
+        {
+            vertices[6],
+            vertices[2],
+            vertices[3],
+            vertices[7]
+        }
+    };
+
+    private static final int[][] initBufs = new int[6][32];
+
+    static {
+        for (ForgeDirection dir : ForgeDirection.VALID_DIRECTIONS) {
+
+            final int dirI = dir.ordinal();
+            int[][] shifts = shiftsByDirection[dirI];
+
+            for (int vi = 0; vi < 4; ++vi) {
+                final int i = vi * 8;
+
+                // XYZ
+                initBufs[dirI][i + 0] = Float.floatToIntBits(shifts[vi][0]);
+                initBufs[dirI][i + 1] = Float.floatToIntBits(shifts[vi][1]);
+                initBufs[dirI][i + 2] = Float.floatToIntBits(shifts[vi][2]);
+
+                // UV
+                initBufs[dirI][i + 3] = Float.floatToIntBits(0);
+                initBufs[dirI][i + 4] = Float.floatToIntBits(0);
+
+                // Color, normal, brightness
+                initBufs[dirI][i + 5] = 0xFFFFFFFF;
+                initBufs[dirI][i + 6] = 0;
+                initBufs[dirI][i + 7] = 15_728_880; // 15 sky 15 block
+            }
+        }
+    }
+
+
+    public static final CubeModel INSTANCE = new CubeModel(false);
+    private final boolean[] colorized = new boolean[6];
+    private final BlockRenderer.Flags flags = new BlockRenderer.Flags(true, true, false, false);
+
+    public CubeModel(boolean colorized) {
+
+        Arrays.fill(this.colorized, colorized);
+    }
+
+    public CubeModel(boolean[] colorized) {
+
+        System.arraycopy(colorized, 0, this.colorized, 0, 6);
+    }
+
+    @Override
+    public List<Quad> getQuads(IBlockAccess world, BlockPos pos, Block block, int meta, ForgeDirection dir, Random random) {
+
+        final Quad face = new Quad();
```

Comment:
Any reason to avoid using the quad pool?  I'm assuming these get allocated and thrown away afterwards?

Reply:
I implemented a quad pool, but from what I can tell unless the writeQuad() implementation releases the quads for reuse, it's the same as just normal allocation. Something to be fixed, perhaps...

Reply:
Yah we'd want to use it in conjunction with release quads... Otherwise no point using the pool it'd grow unbounded.  Worth thinking about, as I don't really want too many allocations in rendering